lib/iris/fileformats/dot.py

```diff
@@ -33,20 +33,35 @@
 _GRAPH_INDENT = ' ' * 4
 _SUBGRAPH_INDENT = ' ' * 8
 
-_DOT_EXECUTABLE_PATH = iris.config.get_option('System', 'dot_path',
-                                              default='dot')
-if not os.path.exists(_DOT_EXECUTABLE_PATH):
-    if not os.path.isabs(_DOT_EXECUTABLE_PATH):
-        try:
-            # Check PATH
-            subprocess.check_output([_DOT_EXECUTABLE_PATH, '-V'],
-                                    stderr=subprocess.STDOUT)
-        except OSError:
-            _DOT_EXECUTABLE_PATH = None
+
+_DOT_CHECKED = False
+_DOT_EXECUTABLE_PATH = None
+
+
+def _dot_path():
+    global _DOT_CHECKED, _DOT_EXECUTABLE_PATH
+
+    if _DOT_CHECKED:
+        path = _DOT_EXECUTABLE_PATH
     else:
-        _DOT_EXECUTABLE_PATH = None
+        path = iris.config.get_option('System', 'dot_path', default='dot')
+        if not os.path.exists(path):
+            if not os.path.isabs(path):
+                try:
+                    # Check PATH
+                    subprocess.check_output([path, '-V'],
+                                            stderr=subprocess.STDOUT)
+                except (OSError, subprocess.CalledProcessError):
+                    path = None
+            else:
+                path = None
+        _DOT_EXECUTABLE_PATH = path
+        _DOT_CHECKED = True
+    return path
+
+
 #: Whether the 'dot' program is present (required for "dotpng" output).
-DOT_AVAILABLE = _DOT_EXECUTABLE_PATH is not None
+DOT_AVAILABLE = _dot_path() is not None
```

Comment:
Surely, with this in place then it is **still making the call at import time** ?

I don't think DOT_AVAILABLE is used anywhere, but then I suppose it _is_ now part of the public API.

If that's important, then its awkward, as you can't wrap a module variable as a "property".
I suppose you could define an object with a custom "bool" conversion ??
It can be done by defining a special method: see http://stackoverflow.com/questions/1087135/boolean-value-of-objects-in-python

Pretty obscure + nasty though...